tests/end-to-end/external-solution-integration/cmd/runner/main.go

```diff
@@ -0,0 +1,88 @@
+package main
+
+import (
+	"github.com/avast/retry-go"
+	"github.com/kyma-project/kyma/tests/end-to-end/external-solution-integration/internal/scenario"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	"os"
+
+	"github.com/kyma-project/kyma/tests/end-to-end/external-solution-integration/pkg/step"
+	log "github.com/sirupsen/logrus"
+	"github.com/spf13/pflag"
+	"k8s.io/cli-runtime/pkg/genericclioptions"
+	coreClient "k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/rest"
+
+	_ "k8s.io/client-go/plugin/pkg/client/auth/gcp"
+)
+
+var scenarios = map[string]scenario.Scenario{
+	"e2e":        &scenario.E2E{},
+	"event-only": &scenario.SendEventAndCheckCounter{},
+}
+
+var (
+	kubeConfig *rest.Config
+	runner     *step.Runner
+)
+
+func main() {
+	scenarioName := os.Args[1]
+	os.Args = os.Args[1:]
+	s, exists := scenarios[scenarioName]
+	if !exists {
+		log.Errorf("Scenario '%s' does not exist. Use one of the following: ", scenarioName)
+		for name := range scenarios {
+			log.Infof(" - %s", name)
+		}
+	}
+
+	runner = step.NewRunner()
+	setupLogging()
+	setupFlags(s)
+	waitForAPIServer()
```

Comment:
Because we have a sidecar. Usually, this is solved by the controller, but here have a script. We need to wait until sidecar is ready and we can reach API server